nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/helpers/ColdStorageHelper.java

```diff
@@ -220,6 +222,19 @@ public static ColdStorageContentStatus checkColdStorageContentAvailability(CoreS
         return new ColdStorageContentStatus(beingRetrieved, available);
     }
 
+    /**
+     * Retrieves the {@link DocumentModel} of the given {@code id}.
+     *
+     * @param session the core session
+     * @param id
+    the document id
+     * @return an optional containing the document model of the given {@code id} if it exists, empty otherwise
+     */
+
+    public static Optional<DocumentModel> getDocumentModelById(CoreSession session, DocumentRef documentIdRef) {
+        return Optional.of(session.getDocument(documentIdRef));
```

Comment:
Here you would want `Optional.ofNullable` if you expect to be able to return an empty `Optional`.
But even this wouldn't work because if the document doesn't exist `session.getDocument` will not return `null` but throw a `DocumentNotFoundException`.
... which means that this code path hasn't been tested.
Finally I don't see the point of this one-line method used only once (which isn't even related to cold storage itself). You should just inline everything in the caller.